src/speechless/benchmark.py

```diff
@@ -0,0 +1,166 @@
+from enum import Enum
+import re
+from rouge import Rouge
+from logging import Logger
+from argparse import ArgumentParser
+
+from speechless.processing.tokenization import EditToken
+from speechless.processing.analysis.tfidf import TfidfAnalysis
+from speechless.utils.logging import NULL_LOGGER
+from speechless.utils.cli import cli_subcommand
+
+
+class TokenGranularity(Enum):
+    WORD = 0
+    SENTENCE = 1
+
+# dataset format: %[bad sentence%], %(bad_word%) 
+class Benchmark:
+
+    def __init__(self, dataset_file : str, method : str, granularity : TokenGranularity, logger: Logger = NULL_LOGGER):
```

Comment:
Let's remove the `method` from the constructor and use a single instance of this class as a benchmark of a single file, with the ability to run it on many methods (add the method as an argument to the `run` function) - you do some preprocessing on the text of the benchmark, which can be done once this way.